src/v/iceberg/table_io.cc

```diff
@@ -19,10 +19,8 @@ ss::future<checked<table_metadata, metadata_io::errc>>
 table_io::download_table_meta(const table_metadata_path& path) {
     return download_object<table_metadata>(
       path(), "iceberg::table_metadata", [](iobuf b) {
-          iobuf_parser p(std::move(b));
-          auto sz = p.bytes_left();
           json::Document parsed;
-          parsed.Parse(p.read_string(sz));
+          parsed.Parse(b.linearize_to_string());
```

Comment:
Hm. These can get past 128k if snapshot retention is not configured/not working correctly (glue).

I'm not sure what I prefer here, failing outright or try to allocate memory and cross fingers

Reply:
@andrwng  see first commit for context

Reply:
let me add a commit to just parse using streaming. Another alternative is to define protobuf and use our new protobufs for serialization. Google has a whole project to convert openapi models to protobuf